### DISCUSSION TITLE
Are there any way to add indentation to first sentence in paragraph and line spacing?

### DISCUSSION THREAD

**Discussion by lverona** [vkbo/novelWriter#1997] [Help] [answered]
Dear Veronica,

This is me asking for help and/or suggesting a formatting feature.

I am also on Linux and after combing through the writing tools available, I think I have landed with novelWriter. I am liking how it looks and it inspires me to write, so thank you so much for developing this app! :pray: 

I have previously used Google Docs. I have to admit, it has a lot of features which are strong - editing history is one. But formatting is another: you can set up indentation options, so that the very first line of a new paragraph is indented, and you can set up line spacing.

I totally appreciate that the idea is that the editing is kept very basic. I was wondering about two things.

1. What do you see about potentially adding line spacing option? I am thinking of purely how novelWriter would display text, as the end result is still a text file with some markdown. But after editors where you can set higher values for line spacing the text in novelWriter looks a bit too dense.

2. If I am importing text from Google Docs, is there any way (or else, could a formatting option be added) that would add a tab in front of the first sentence of every paragraph?

Thank you again for all your work!

**Comment by vkbo** [comment, marked as answer]
You can set both line spacing and first line indentation when you build the manuscript from the manuscript tool (F5). In the editor, you cannot change these settings. It is unfortunately not possible to change line height in the plain text editor novelWriter is built on. You can see the feature request here: #654. If it at some point becomes possible, I will definitely add it.

The plain text editor that is used here is locked to the font height, so if you can find a font with more spacing, you can get around the problem. Unfortunately, that is the only option right now.

There is no way to insert tabs automatically now. Since it is plain text, paragraphs are separated by a blank line, as is the convention in most styles. I'm working on a side project with an editor similar to novelWriter, but with rich text editing at the core, where these formatting options are available. But I don't really have the time for two projects right now, so it's going slow.

As for editing history, I've been wanting to add something for that for a long time (see #383), but I haven't found a good way of solving it yet.

**Reply by lverona** [reply]
Thank you! Makes sense!

I might simply get used to seeing the text that way, that's not a problem. The tab is a slightly bigger problem because it's just very difficult to understand where the new paragraph starts, and common formatting for books doesn't favor spaces between paragraphs.

**Reply by vkbo** [reply]
Yes, but this is just a limitation in the editor itself. The manuscript tool will generate a manuscript of your novel with the paragraph formatting you're looking for. You have to select Open Document or HTML output for it to work, as it isn't applied to Markdown output.

**Reply by lverona** [reply]
Yes, this is no problem! I am not worried about the end manuscript, only about how it looks as I'm writing it.

**Reply by vkbo** [reply]
Ok, I see. I too prefer the indented paragraphs, but it isn't the convention for plain text. so I too had to get used to it. I'm doing it differently in my other editor project, but as I mentioned, I don't have enough time to work on it. One Day™ maybe ... 😄

**Reply by lverona** [reply]
It's not a problem when I'm writing a new project, of course, so no big deal. I'm just tabbing every new paragraph, which is fine.

As for the new project, as a developer myself I know the feeling. This could be too much, and, honestly, you are a savior for us, Linux folks! <3
